pkg/binlog/reader/util.go

```diff
@@ -60,6 +63,16 @@ func GetGTIDsForPos(ctx context.Context, r Reader, endPos gmysql.Position, parse
 		// NOTE: only update endPos/GTIDs for DDL/XID to get an complete transaction.
 		switch ev := e.Event.(type) {
 		case *replication.QueryEvent:
+			parser2 := parser.New()
+			ansiQuotes, err := event.GetAnsiQuotesMode(ev.StatusVars)
+			if err != nil {
+				log.L().Warn("can't determine ANSI_QUOTES from binlog status_vars, use no ANSI_QUOTES instead", zap.Error(err))
+				ansiQuotes = false
+			}
+			if ansiQuotes {
+				parser2.SetSQLMode(tmysql.ModeANSIQuotes)
```

Comment:
I may as well parse all available SQL mode and set here 😂 

will check "Full List of SQL Modes" in https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html tomorrow